src/databricks/labs/ucx/source_code/linters/directfs.py

```diff
@@ -141,26 +147,88 @@ def lint_tree(self, tree: Tree) -> Iterable[Advice]:
             )
             yield advisory
 
+    def collect_dfsas(self, python_code: str, inherited_tree: Tree | None) -> Iterable[DirectFsAccessNode]:
+        tree = Tree.new_module()
+        if inherited_tree:
+            tree.append_tree(inherited_tree)
+        tree.append_tree(Tree.normalize_and_parse(python_code))
+        visitor = _DetectDirectFsAccessVisitor(self._session_state, self._prevent_spark_duplicates)
+        visitor.visit(tree.node)
+        yield from visitor.directfs_nodes
 
-class DirectFsAccessSqlLinter(SqlLinter):
 
-    def lint_expression(self, expression: Expression):
-        for table in expression.find_all(Table):
-            # Check table names for direct file system access
-            yield from self._check_dfsa(table)
+class DirectFsAccessSqlLinter(SqlLinter):
```

Comment:
is it called from `DirectFsAccessPyLinter` for `spark.sql(XXX)` case? if not - it has to.

Reply:
No. We're not linting query in `spark.sql`. That seems to be an old bug and much broader bug. I suggest fixing that in a separate PR (with missing tests). See https://github.com/databrickslabs/ucx/issues/2551

Reply:
See PR #2558